src/web/command_control/client/components/CommandControl.tsx

```diff
@@ -1052,11 +1064,33 @@ export default class CommandControl extends React.Component {
 	/**
 	 * Updates the runList in state
 	 * 
-	 * @param {MissionInterface} runList updated version used to set state
+	 * @param {MissionInterface} runList Updated version used to set state
 	 * @returns {void}
 	 */
 	setRunList(runList: MissionInterface) {
-		this.setState({runList, runListVersion: this.state.runListVersion + 1})
+		const updatedRunList = {...runList}
+		this.setState({
+			runList: updatedRunList, 
+			runListVersion: this.state.runListVersion + 1
+		})
+	}
+
+	/**
+	 * Saves a deep copy of the current mission state so the operator can restore changes
+	 * 
+	 * @param {MissionInterface} mission Provides current mission data to append to mission history
+	 * @returns {void}
+	 * 
+	 * @notes
+	 * The maximum size of the missionHistory array is set to the variable missionHistoryMaxLen.
+	 * This is to prevent the mission history from causing performance issues.
```

Comment:
Maybe we can be more explicit why it would cause an issue:
This is to prevent the mission history from causing performance issues -> 
This is to prevent the mission history from growing unbounded and potentially causing performance issues